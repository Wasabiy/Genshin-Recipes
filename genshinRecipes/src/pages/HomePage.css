video {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  object-fit: cover;
  z-index: -1;
}

#wanminTitle {
  border-right: solid 0.5vh rgba(252, 189, 0, 0.75);
  white-space: nowrap;
  overflow: hidden;
  display: flex;
  flex-wrap: wrap;
  font-family:
    OCR A Std,
    monospace;
  font-size: 2.7vh;
  color: rgba(255, 255, 255, 0.7);
  margin: 0 auto;
  animation:
    animated-text 5s steps(35, end) infinite normal both,
    animated-cursor 600ms steps(35, end) infinite;
}

#homeTitle {
  margin-top: 10vh;
  padding: 10px;
  display: flex;
  flex-wrap: wrap;
}

@keyframes animated-text {
  from {
    width: 0;
  }
  to {
    width: 110vh;
  }
}

@keyframes animated-cursor {
  from {
    border-right-color: rgba(252, 189, 0, 0.75);
  }
  to {
    border-right-color: transparent;
  }
}

@keyframes pause-text {
  from {
    opacity: 1;
  }
  to {
    opacity: 1;
  }
}
