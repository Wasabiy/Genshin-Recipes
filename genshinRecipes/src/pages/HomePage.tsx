import homePageMovieMp4 from '../assets/homepage.mp4';
import homePageMovieWebm from '../assets/homepage.webm';
import './HomePage.css';
import { useNavigate } from 'react-router-dom';

function HomePage() {
  const navigateRecipes = useNavigate();

  const handleClick = (e: { stopPropagation: () => void }) => {
    e.stopPropagation();
    navigateRecipes('/recipes');
  };

  return (
    <>
      <section id="WholePage">
        <video loop autoPlay muted>
          <source src={homePageMovieMp4} type="video/mp4" />
          <source src={homePageMovieWebm} type="video/webm" />
        </video>
        <section id="homeTitle">
          <h1 id="wanminTitle" onClick={handleClick}>
            Welcome to Wanmin Restaurant. Press on this text to see more............
          </h1>
        </section>
      </section>
    </>
  );
}
export default HomePage;
