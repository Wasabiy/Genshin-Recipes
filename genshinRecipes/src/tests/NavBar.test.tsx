import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import { userEvent } from "@testing-library/user-event";
import { describe, expect, test } from "vitest";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import NavBar from "../components/Navbar.tsx";
import HomePage from "../pages/HomePage.tsx";

describe("NavBar.tsx snapshot test", () => {
    test('matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("NavBar navigation testing", () => {

    test("Check to see if NavBar will navigate to HomePage.tsx when 'The Wanmin Restaurant' is clicked.", async () => {
        render(
            <MemoryRouter initialEntries={["/"]}>
              <NavBar />
              <Routes>
                <Route path="/" element={<HomePage />} />
              </Routes>
            </MemoryRouter>
          );
        const wanminLink = screen.getByText('The Wanmin Restaurant'); // Find the link with text 'Recipes'
        expect(wanminLink).toBeInTheDocument(); // Ensure it's in the document

        await userEvent.click(wanminLink);
        expect(screen.getByText("The Wanmin Restaurant")).toBeInTheDocument();
})
})