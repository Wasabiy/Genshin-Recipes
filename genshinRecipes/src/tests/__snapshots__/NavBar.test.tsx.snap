// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar.tsx snapshot test > matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="homePage"
  >
    <div
      class="navbar-logo"
    >
      <a
        href="/"
      >
        The Wanmin Restaurant
      </a>
    </div>
    <ul
      class="navbar-links"
    >
      <li>
        <a
          href="/recipes"
        >
          <img
            alt="primo"
            id="textLogo"
            src="/project1/src/assets/genshinFood.png"
          />
          Recipes
        </a>
      </li>
      <li>
        <a
          href="/favorites"
        >
          <img
            alt="primo"
            id="textLogo"
            src="/project1/src/assets/primogem.png"
          />
          My Favorites
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          <img
            alt="paimon"
            src="/project1/src/assets/paimon.png"
          />
          About
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
